### PR TITLE
test: stats payload tests

### PR DESCRIPTION
**Pull request by merlinstardust**

### CODE REVIEW

File: tests/functional/spec/payload.test.js [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+        const requests = await runTest({
+            config,
+            testPage,
+            callback: async ({ bannerFrame, modalFrame }) => {
+                await clickBanner(bannerFrame);
+                await modalFrame.click(selectors.button.btn);
+            }
+        });
+
+        const request = requests.find(r => r.bdata.link && request.bdata.link.includes('Apply Now'));
```
Comment (gillycheesesteak) [inline tests/functional/spec/payload.test.js]: I think this should be `r.bdata.link.includes()`

File: tests/functional/jest.config.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 module.exports = {
-    testMatch: ['<rootDir>/**/?(*.)test.js?(x)'],
+    testMatch: ['<rootDir>/**/?(*.)test.js?(x)', '<rootDir>/spec/payload.test.js'],
```
Comment (gillycheesesteak) [inline tests/functional/jest.config.js]: Does the first match pattern not catch this file?

Comment (merlinstardust) [reply tests/functional/jest.config.js]: For some reason, this was the only way I could catch this file. I feel like it should match the pattern of the first one but it skips it.

File: package.json [hunk 3]
```diff
@@ -74,6 +74,8 @@
         "prettier": "^1.14.3",
         "prettier-eslint": "^9.0.1",
         "puppeteer": "^2.0.0",
+        "puppeteer-request-spy": "^1.3.1",
```
Comment (Seavenly) [inline package.json]: Do we need to include this obscure package? Or can we accomplish what we need with a few lines of code using the Puppeteer API?

Comment (merlinstardust) [reply package.json]: It might be possible but I'm not sure how. My research pointed me to this as the quickest way to see requests that are sent out.

Comment (Seavenly) [reply package.json]: It looks like we're trying to capture certain requests based on keyword matching. Could this similarly help to accomplish it? https://pptr.dev/#?product=Puppeteer&version=v2.1.1&show=api-event-request

Comment (merlinstardust) [reply package.json]: Fixed finally!

**Pull request merged**